SendWithUs.Client.Tests/Component/JsonObjectWriter.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Copyright © 2015 Mimeo, Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+namespace SendWithUs.Client.Tests.Component
+{
+    using System;
+    using System.IO;
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+
+    public class JsonObjectWriter : JsonTextWriter

```

Comment:
Maybe just JsonStringWriter since that's what you've really done from the "writer" perspective, you've further specialized from `TextWriter` to `StringWriter`. I only wonder if `WithCache` is good because that's kind of abstract. You're not really caching the object per-se, you're just buffering its JSON string representation.

It's a weird class to me cause it's actually for reading, but you've inherited from something that's for writing just so you don't have to manage the buffer separately. So I question it's overall design, it's just a convenience to not have to manage the buffer separately, but is it worth the confusion of creating something that now seemingly has dual/conflicting responsibilities?

Again, this is tests not like part of the library design, so it's not *that* important to me. Just thought worth a discussion.